src/main/scala/com.snowplowanalytics.iglu/ctl/commands/TableCheck.scala

```diff
@@ -0,0 +1,263 @@
+    for {
+      schemaJsons <- Stream.eval(Pull.getSchemas(Server.buildPullRequest(registryRoot, readApiKey)))
+      schemas     <- Stream.eval(EitherT.fromEither[IO](Generate.parseSchemaJsonsToSchemas(schemaJsons)))
+      modelGroupMap = createModelGroups(schemas)
+      temp        <- Stream.emits[Failing, (SchemaKey, List[IgluSchema])](modelGroupMap.toList)
+      (schemaKey, schemaModelGroup) = temp
+      res         <- Stream.eval(checkTable(storage, schemaKey, schemaModelGroup, dbschema))
+    } yield res
+
+  /**
+    * Fetches model group schemas of given schemaKey with resolver which is created
+    * from given resolver config
+    */
+  def fetchSchemaModelGroup(resolverPath: Path, schemaKey: SchemaKey): Failing[List[IgluSchema]] =
+    for {
+      resolverJson <- EitherT(File.readFile(resolverPath).map(_.flatMap(_.asJson)))
+      resolver     <- EitherT(Resolver.parse[IO](resolverJson.content))
+        .leftMap(e => Error.ConfigParseError(s"Resolver can not created: $e"))
+      schemaJsons  <- resolver.fetchSchemas(schemaKey.vendor, schemaKey.name, Some(schemaKey.version.model))
+        .leftMap(e => Error.ServiceError(s"Error while lookup for schema: ${(e: ClientError).asJson.noSpaces}"))
+        .map(_.filter(s => Ordering[SchemaVer].gteq(schemaKey.version, s.self.schemaKey.version)))
+      schemas      <- EitherT.fromEither[IO](Generate.parseSchemaJsonsToSchemas(schemaJsons))
+    } yield schemas
+
+  /**
+    * Creates table structure with given schemas
+    */
+  def buildTableDdl(schemas: List[IgluSchema]): Either[Common.Error, List[DDLColumn]] =
+    NonEmptyList.fromList(schemas) match {
+      case None => List.empty[DDLColumn].asRight
+      case Some(nonEmptySchemas) =>
+        val orderedSubSchemaMap = Migration.buildOrderedSubSchemasMap(nonEmptySchemas)
+        val res = orderedSubSchemaMap.toList match {
+          case (_, orderedSubSchemas)::Nil =>
+            DdlGenerator.generateTableDdl(orderedSubSchemas, "", None, 4096, false).asRight[Common.Error]
+          case Nil =>
+            Error.Message("There is no ordered subschemas to generate table ddl").asLeft[CreateTable]
+          case l =>
+            Error.Message(s"There must be only one ordered subschemas to generate table ddl but there are multiple: ${l.map(_._1)}").asLeft[CreateTable]
+        }
+        res.map(_.columns)
+    }
+
+  /**
+    * Compares existing and expected columns whether they are same or not.
+    */
+  private[ctl] def checkColumns(schemaKey: SchemaKey, existingColumns: List[Column], expectedColumns: List[DDLColumn]): TableCheckResult =
+    existingColumns match {
+      case Nil => TableNotDeployed(schemaKey)
+      case _ =>
+        expectedColumns.map(_.columnName).zipAll(existingColumns.map(_.columnName), "", "").foldLeft(List.empty[(String, String)]) {
+          case (acc, (expected, existing)) =>
+            if (!expected.equals(existing)) (existing, expected) :: acc else acc
```

Comment:
Why not `!=`?

Also, I would personally rewrite it with two cases:

Also, a super-nit, there's a big advantage in making some functions as generic as possible. It alleviates local reasoning and testing. I'd write a `dropEqual` function:

```scala
def dropEqual[A: Eq: Monoid](a: List[A], b: List[A]) =
  a.zipAll(b, Monoid[A].empty, Monoid[A].empty)
   .foldLeft(List.empty[(A, A)]) {
     case (acc, (aa, bb)) if aa === bb => acc
     case (acc, (aa, bb))              => (aa, bb) :: acc
   }
```

The same function, but completely detached from column names, strings, other types or other kinds of surrounding information. You don't have to change it here, but that's a nice trick to increase testability and readability - not a big win when we work with `String`s, but if it were `CollectorConfigMetadata` containing of tens of members it would be a lot easier to work with a generic version of the function.